src/appleseed.studio/mainwindow/mainwindow.cpp

```diff
@@ -1268,26 +1295,8 @@ void MainWindow::slot_save_project_as()
             m_settings,
             SETTINGS_FILE_DIALOG_PROJECTS);
 
-    if (!filepath.isEmpty())
-    {
-        const QString Extension = "appleseed";
-
-        if (QFileInfo(filepath).suffix() != Extension)
-            filepath += "." + Extension;
-
-        filepath = QDir::toNativeSeparators(filepath);
-
-        if (m_project_file_watcher)
-            stop_monitoring_project_file();
-
-        m_project_manager.save_project_as(filepath.toAscii().constData());
-
-        if (m_project_file_watcher)
-            start_monitoring_project_file();
-
-        update_recent_files_menu(filepath);
-        update_workspace();
-    }
+    save_project(filepath);
```

Comment:
There is a regression here: if the user cancelled the *Save As* dialog, `filepath` is empty. It appears like this case is no longer handled correctly.